src/model.ts

```diff
@@ -39,14 +39,19 @@ import { cloneDeep } from "./util/clonedeep"
 import { nonenumerable } from "./util/decorators"
 import { IncludeScopeHash } from "./util/include-directive"
 
+export interface KeyCase {
+  from: "dash" | "camel" | "snake"
```

Comment:
may want to extract these strings to a separate type for easier reuse: `type KeyCaseValue = "dash" | "camel" | "snake"`